pkg/ccip/provider/provider.go

```diff
@@ -20,26 +24,40 @@ var _ commontypes.CCIPProvider = &Provider{}
 const CCIPProviderName = "TONCCIPProvider"
 
 type Provider struct {
-	lggr logger.Logger
-	ca   ccipocr3.ChainAccessor
-	ct   ocr3types.ContractTransmitter[[]byte]
+	lggr  logger.Logger
+	ca    ccipocr3.ChainAccessor
+	ct    ocr3types.ContractTransmitter[[]byte]
+	codec ccipocr3.Codec
 
 	wg sync.WaitGroup
 	services.StateMachine
 }
 
-func NewCCIPProvider(lggr logger.Logger, txm txm.TxManager) (*Provider, error) {
+func NewCCIPProvider(lggr logger.Logger, chainSelector ccipocr3.ChainSelector, client ton.APIClientWrapped, txm txm.TxManager, logPoller logpoller.LogPoller) (*Provider, error) {
 	ct, err := ocr.NewCCIPTransmitter(txm, lggr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter %w", err)
+		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter: %w", err)
 	}
 
-	cp := &Provider{
-		lggr: logger.Named(lggr, CCIPProviderName),
-		ct:   ct,
+	codec := ccipocr3.Codec{
```

Comment:
If we are moving core plugin components from capability/ccip/ccipxxx to chain specific repo ? We would need to msghasher and gas_helpers (maybe cr/cw config as well)

Reply:
We want to extract everything from core, I think some of the types were left behind when we did our first pass but if it's not a lot of effort we should go ahead and do it